### ISSUE TITLE
Show current function name

### ISSUE THREAD

**Issue by vmiklos** [rhysd/libclang-vim#5]
Hi,

Maybe it's already possible to do this, but I could not figure it out. So vim has partial support for showing the current function you're in (even if the cursor is not at the function declaration), like this:

http://vim.wikia.com/wiki/Show_current_function_name_in_C_programs

But needless to say if you have complex enough code (attributes, namespaces, etc), then it fails to do the right thing. That's why I think it would be nice if clang could be used to provide that information.

I can show the type under the cursor with e.g.

```
:echo libclang#deduction#type_at(expand('%:p'), line('.'), col('.'), '-std=c++1y').type
```

But I couldn't find an equivalent to print the current function name. I guess it's obvious, but I mean like in your example place the cursor on `std::cout` and the result would be `main`. (In C++ code gdb prints `namespace::Class::member_function`, so probably providing that would be the best.)

To sum up, if this already works, then it's not obvious from the README. :-)

Thanks,

Miklos

**Comment by vmiklos** [comment]
Here is a self-contained PoC doing this:

https://github.com/vmiklos/vmexam/blob/master/libclang/get-current-function.cxx

I'll try to look into integrating this into libclang-vim properly.

**Comment by rhysd** [comment]
Thank you for the implementation.  I merged the PR.

I haven't developed this repository since clang 3.5 because I have fewer chances to use C++.  If you're interested in this repository, do you want to be a committer?  I can add you to collaborators.

**Comment by vmiklos** [comment]
Sure, that would be great! Thanks. :-)

**Comment by rhysd** [comment]
I added you to collaborators :smile:
